lib/routes/shu/index.ts

```diff
@@ -28,50 +25,67 @@ export const route: Route = {
     },
     radar: [
         {
-            source: ['www.shu.edu.cn/:type'],
-            target: '/:type',
+            source: ['www.shu.edu.cn/'],
+            target: '/news',
         },
     ],
-    name: '官网信息',
-    maintainers: ['lonelyion'],
+    name: '官网通知公告',
+    maintainers: ['GhhG123', 'lonelyion'],
     handler,
-    description: `| 综合新闻 | 科研动态 | 通知公告 | 重要新闻  |
-  | -------- | -------- | -------- | --------- |
-  | news     | research | notice   | important |`,
+    url: 'www.shu.edu.cn/',
+    description: `| 通知公告 | 重要新闻 |
+  | -------- | --------- |
+  | tzgg     | zyxw      |`,
 };
 
 async function handler(ctx) {
-    const type = ctx.req.param('type') || 'news';
-    const link = `https://www.shu.edu.cn/${alias.get(type) || type}.htm`;
-    const respond = await got.get(link);
-    const $ = load(respond.data);
-    const title = $('title').text();
-    const list = $('.ej_main .list')
-        .find('li')
-        .slice(0, 5)
-        .toArray()
-        .map((ele) => ({
-            title: $(ele).find('.bt').text(),
-            link: new URL($(ele).find('a').attr('href'), host).href,
-            date: $(ele).find('.sj').text(),
-        }));
+    const type = ctx.req.param('type') ?? 'tzgg';
+    const rootUrl = 'https://www.shu.edu.cn';
+
+    // 发起 HTTP GET 请求
+    const response = await got({
+        method: 'get',
+
+        /* headers: {
+            'user-agent': UA,
+            cookie: await getCookie(ctx),
+        }, */
+        url: notice_type[type].url,
+    });
+
+    const $ = load(response.data);
 
-    const all = await Promise.all(
+    const list = $('div.list ul li') // 以下获取信息需要根据网页结构定制
+        // For cheerio 1.x.x . The item parameter in the .map callback is now explicitly typed as a Cheerio<Element>, not just Element. --fixed
+        .map((_, el) => {
+            const item = $(el); // Wrap `el` in a Cheerio object
+            const rawLink = item.find('a').attr('href');
+            return {
+                title: item.find('p.bt').text().trim(),
+                link: rawLink ? new URL(rawLink, rootUrl).href : rootUrl,
+                pubDate: timezone(parseDate(item.find('p.sj').text().trim(), 'YYYY.MM.DD'), +8),
+                description: item.find('p.zy').text().trim(),
+            };
+        })
+        .get();
+
+    const items = await Promise.all(
         list.map((item) =>
             cache.tryGet(item.link, async () => {
-                const response = await got.get(item.link);
-                const $ = load(response.data);
-                item.author = $('.xx>:nth-child(2)').text().trim().slice(3); // 投稿：xxx
-                item.pubDate = parseDate(item.date, 'YYYY.MM.DD');
-                item.description = $('.v_news_content').html() || item.title;
+                const detailResponse = await got(item.link);
```

Comment:
Use https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/index.ts#L46-L54 or https://github.com/DIYgod/RSSHub/blob/8b48453ed55165a9e2adbda0705bb8bf6accd1e1/lib/routes/shu/index.ts#L75 but not both style